src/libcore/num/mod.rs

```diff
@@ -2504,6 +2504,30 @@ impl fmt::Display for TryFromIntError {
     }
 }
 
+#[unstable(feature = "try_from", issue = "33417")]
+impl TryFrom<bool> for bool {
+    type Error = ();
```

Comment:
The real solution to this is an eventual `impl<A, B> TryFrom<A> for B where B: From<A>` with `type Error = !;`, but that's blocked for several reasons.